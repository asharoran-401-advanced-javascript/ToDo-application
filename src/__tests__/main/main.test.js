/* eslint-disable no-unused-vars */
// eslint-disable-next-line strict
'use strict';

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Main from '../../component/mainToDo/main.js';

describe( '<Main />' , () => {
  it('Main should exist at Todo application' , () =>{
    let app = shallow(<Main />);
    expect(app.find('.todo').exists).toBeTruthy();
    expect(app.find('.difficulty').exists).toBeTruthy();
    expect(app.find('.form-button').exists).toBeTruthy();
    expect(app.find('.item').exists).toBeTruthy();
    expect(app.find('.todo-details').exists).toBeTruthy();
  });

  //   it('Main should contain the todo form and todo list and item details' ,() => {
  //     let app = mount(<Main />);
  //     let todo = app.find('.todo');
  //     let difficulty = app.find('.difficulty');
  //     let formButton = app.find('.form-button');
  //     let item = app.find('.item');
  //     let details = app.find('.todo-details');
  //   });
  //renderer is a package thatâ€™s able to take a React component and render it as a pure JavaScript object.
  it('render correctly' , () => {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});