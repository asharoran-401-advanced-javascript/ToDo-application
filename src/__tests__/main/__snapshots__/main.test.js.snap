// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render correctly 1`] = `
<h1>
  My ToDo Apllication
</h1>
`;

exports[`<Main /> render correctly 1`] = `
<section
  className="todo"
>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        <span>
          ToDo Item :
        </span>
        <input
          name="text"
          onChange={[Function]}
          placeholder="Add Item"
        />
      </label>
      <label>
        <span>
          Difficulty :
        </span>
        <input
          className="difficulty"
          defaultValue="3"
          max="5"
          min="1"
          name="difficulty"
          onChange={[Function]}
          type="range"
        />
      </label>
      <label>
        <span>
          Assigned To :
        </span>
        <input
          name="assignee"
          onChange={[Function]}
          placeholder="Assigned To"
          type="text"
        />
      </label>
      <label>
        <span>
          Date :
        </span>
        <input
          name="Time"
          onChange={[Function]}
          type="date"
        />
      </label>
      <button
        className="form-button"
      >
        Add Item
      </button>
    </form>
  </div>
  <div>
    <ul />
  </div>
</section>
`;
